libsolidity/formal/BMC.cpp

```diff
@@ -47,12 +47,13 @@ BMC::BMC(
 	smt::EncodingContext& _context,
 	UniqueErrorReporter& _errorReporter,
 	UniqueErrorReporter& _unsupportedErrorReporter,
+	ErrorReporter& _provedSafeReporter,
```

Comment:
Instead of using a separate `ErrorReporter` for the safe targets, cannot we just turn `UniqueErrorReporter _errorReporter` into a regular `ErrorReport` and use only that?

Reply:
Right. That's a mistake.

Reply:
The reason we have multiple reporters here is that we want to squash certain kinds of warnings into a single message by default, but also have an option to display them all, depending on the settings. See #14028 / #11597.

Not sure if that was the intention behind `_provedSafeReporter` as well. If not, those errors could just go to `_errorReporter` along with everything else.

Reply:
I understand the reason why we want to avoid duplicated "unsupported" warnings, so `_unsupportedErrorReporter` definitely should stay `UniqueErrorReporter`.
On the other hand, there can be multiple targets (underflow + overflow) for the same expression, and we do not want to lose reports on these, so `UniqueErrorReporter` should not be used for those. I would like to investigate if the error report for all the other things has to be unique or not.